usersession/client/client.go

```diff
@@ -298,27 +300,108 @@ func (client *Client) serviceControlCall(ctx context.Context, action string, ser
 	return startFailures, stopFailures, err
 }
 
+func (client *Client) serviceControlCall(ctx context.Context, inst *ServiceInstruction) (startFailures, stopFailures []ServiceFailure, err error) {
+	headers := map[string]string{"Content-Type": "application/json"}
+	reqBody, err := json.Marshal(inst)
+	if err != nil {
+		return nil, nil, err
+	}
+	responses, err := client.doMany(ctx, "POST", "/v1/service-control", nil, headers, reqBody)
+	if err != nil {
+		return nil, nil, err
+	}
+	return client.decodeControlResponses(responses)
+}
+
 func (client *Client) ServicesDaemonReload(ctx context.Context) error {
-	_, _, err := client.serviceControlCall(ctx, "daemon-reload", nil)
+	_, _, err := client.serviceControlCall(ctx, &ServiceInstruction{
+		Action: "daemon-reload",
+	})
 	return err
 }
 
-func (client *Client) ServicesStart(ctx context.Context, services []string) (startFailures, stopFailures []ServiceFailure, err error) {
-	return client.serviceControlCall(ctx, "start", services)
+func filterDisabledServices(all, disabled []string) []string {
+	var filtered []string
+ServiceLoop:
+	for _, svc := range all {
+		for _, disabledSvc := range disabled {
+			if strings.Contains(svc, disabledSvc) {
+				continue ServiceLoop
+			}
+		}
+		filtered = append(filtered, svc)
+	}
+	return filtered
 }
 
-func (client *Client) ServicesStop(ctx context.Context, services []string) (stopFailures []ServiceFailure, err error) {
-	_, stopFailures, err = client.serviceControlCall(ctx, "stop", services)
-	return stopFailures, err
+func (client *Client) ServicesStart(ctx context.Context, services []string, disabledSvcs map[int][]string, enable bool) (startFailures, stopFailures []ServiceFailure, err error) {
```

Comment:
Since you're already modifying code in this file, why not add a short doc comment near the exported functions in the chunks that you modified.

Reply:
Yea that makes a lot of sense, thank you, added!